_pytest/cacheprovider.py

```diff
@@ -14,7 +14,8 @@
 class Cache(object):
     def __init__(self, config):
         self.config = config
-        self._cachedir = config.rootdir.join(".cache")
+        cache_dirname = config.getini("cache_dirname")
+        self._cachedir = config.rootdir.join(cache_dirname)
```

Comment:
It was also my understanding that this was supposed to be relative to the rootdir.

I'm curious, why do you guys say it would be more difficult to support absolute paths? Can't we assume:

1. If relative path, relative to `rootdir`;
2. If absolute path, just use it as it is;

I think this is a interim solution for #1089 for example. We might even consider calling `os.path.expandvars` on the variable, which would allow users to specify `$XDG_CONFIG_DIR` as the cache directory.